File: src/com/mishiranu/instantimage/util/FileManager.java
package com.mishiranu.instantimage.util;

import java.io.File;
import java.util.ArrayList;

import android.content.Context;
import android.webkit.MimeTypeMap;

public class FileManager
{
	public static void cleanup(Context context)
	{
		File[] files = context.getFilesDir().listFiles();
		if (files != null)
		{
			long time = System.currentTimeMillis() - 24 * 60 * 60 * 1000;
			for (File file : files)
			{
				if (file.lastModified() < time) file.delete();
			}
		}
	}
	
	public static class FileItem
	{
		public final int contentId;
		public final String displayName;
		public final File file;
		
		public FileItem(int contentId, String displayName, File file)
		{
			this.contentId = contentId;
			this.displayName = displayName;
			this.file = file;
		}
		
		public String getMimeType()
		{
			String extension = null;
			int index = displayName.lastIndexOf('.');
			if (index >= 0) extension = displayName.substring(index + 1);
			if (extension == null || extension.isEmpty()) extension = "jpeg";
			String mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(extension);
			if (mimeType == null) mimeType = "image/jpeg";
			return mimeType;
		}
	}
	
	public static String obtainSimpleFileName(String originalUriString)
	{
		String extension = null;
		if (originalUriString != null)
		{
			int index = originalUriString.lastIndexOf('.');
			if (index >= 0 && originalUriString.indexOf('/', index) == -1)
			{
				String extensionTemp = originalUriString.substring(index + 1);
				if (extensionTemp.length() <= 5) extension = extensionTemp;
			}
		}
		if (extension == null || extension.isEmpty()) extension = "jpeg";
		return System.currentTimeMillis() + "." + extension;
	}
	
	public static FileItem obtainFile(Context context, String originalUriString, int contentId)
	{
		File filesDir = context.getFilesDir();
		filesDir.mkdirs();
		String displayName = contentId + "-" + obtainSimpleFileName(originalUriString);
		return new FileItem(contentId, displayName, new File(filesDir, displayName));
	}
	
	public static ArrayList<FileItem> listFiles(Context context)
	{
		ArrayList<FileItem> fileItems = new ArrayList<>();
		File[] files = context.getFilesDir().listFiles();
		if (files != null)
		{
			for (File file : files)
			{
				String[] splitted = file.getName().split("-");
				if (splitted.length == 2)
				{
					fileItems.add(new FileItem(Integer.parseInt(splitted[0]), splitted[1], file));
				}
			}
		}
		return fileItems;
	}
	
	public static FileItem findFile(Context context, int contentId)
	{
		ArrayList<FileItem> fileItems = listFiles(context);
		for (FileItem fileItem : fileItems)
		{
			if (fileItem.contentId == contentId) return fileItem;
		}
		return null;
	}
}

File: src/com/mishiranu/instantimage/util/FileManager.scala
package com.mishiranu.instantimage.util

import java.io.File

import android.content.Context
import android.webkit.MimeTypeMap

object FileManager {
  def cleanup(context: Context): Unit = {
    val time = System.currentTimeMillis - 24 * 60 * 60 * 1000
    Option(context.getFilesDir.listFiles).getOrElse(Array())
      .filter(_.lastModified < time)
      .foreach(_.delete())
  }

  def obtainSimpleFileName(originalUriString: String): String = {
    System.currentTimeMillis + "." + Option(originalUriString).map { uriString =>
      val index = uriString.lastIndexOf('.')
      if (index >= 0 && uriString.indexOf('/', index) == -1) uriString.substring(index + 1) else null
    }.filter(e => e != null && !e.isEmpty && e.length <= 5).getOrElse("jpeg")
  }

  case class FileItem(contentId: Int, displayName: String, file: File) {
    def mimeType: String = {
      val extension = {
        val index = displayName.lastIndexOf('.')
        if (index >= 0) displayName.substring(index + 1) else "jpeg"
      }
      Option(MimeTypeMap.getSingleton.getMimeTypeFromExtension(extension))
        .filter(_ != null)
        .filter(_.isEmpty)
        .getOrElse("image/jpeg")
    }
  }

  def obtainFile(context: Context, originalUriString: String, contentId: Int): FileItem = {
    val filesDir = context.getFilesDir
    filesDir.mkdirs()
    val displayName = contentId + "-" + obtainSimpleFileName(originalUriString)
    FileItem(contentId, displayName, new File(filesDir, displayName))
  }

  def listFiles(context: Context): List[FileItem] = {
    case class Data(file: File, splitted: Array[String])
    Option(context.getFilesDir.listFiles).getOrElse(Array())
      .map(f => Data(f, f.getName.split("-")))
      .filter(_.splitted != null)
      .filter(_.splitted.length == 2)
      .map(d => FileItem(d.splitted(0).toInt, d.splitted(1), d.file)).toList
  }

  def findFile(context: Context, contentId: Int): Option[FileItem] = {
    listFiles(context).find(_.contentId == contentId)
  }
}
